### DISCUSSION TITLE
Installazione Home Assistant e primo accesso

### DISCUSSION THREAD

**Discussion by Al3eRtO** [indomus/forum#1889] [GENERALE - Discussioni generiche]
Buongiorno a tutti, ho seguito questa guida per installare home assistant:

https://indomus.it/guide/installare-e-configurare-home-assistant-core-con-docker-su-mini-pc-intel-nuc-computer-con-linux-debian/

Nell'ordine:
Linux Debian  su laptop senza altri os: https://indomus.it/guide/installare-mini-pc-intel-nuc-con-linux-debian-11-bullseye/
Installato Docker: https://indomus.it/guide/come-installare-docker-su-linux-debian/
Installato Home Assistant tramite comando “docker run”
Primo accesso a Home Assistant:
http://homeassistant.local:8123/ non funziona
http://INDIRIZZO_IP_DEL_SERVER:8123 non funziona
Ho verificato che l'ip sia corretto, che prove potrei fare per capire cosa non funziona?
Grazie

**Comment by snakuzzo** [comment]
Ciao, il container è up & running ?
Hai almeno visto i comandi base di docker ?

**Reply by Al3eRtO** [reply]
Credo di sì, da guida la verifica mi da il risultato:

Unable to find image 'hello-world:latest' locally
latest: Pulling from hello-world
a0691bf12e4e: Pull complete 
Digest: sha256:9701edc932223a66e49dd6c894a11db8c2cf4eccd1414f1ec105a623bf16b426
Status: Downloaded newer image for hello-world:latest

Hello from Docker!
This message shows that your installation appears to be working correctly.

**Reply by snakuzzo** [reply]
Sì ma questo è il container di hello world.
Devi verificare che il container di home assistant sia stato creato e stia girando.

**Reply by Al3eRtO** [reply]
Allora no, la guida dice di:

Istanziamo dunque Home Assistant Core su Docker tramite il comando:

"docker run --init -d --name="home-assistant" -e "TZ=Europe/Rome" -v "/home/deb/homeassistant:/config" -v "/run/dbus:/run/dbus:ro" --net=host --restart always ghcr.io/home-assistant/home-assistant:stable"

Lanciato il comando, attenderne il completamento. Ci potrebbe volere un po’.

N.b. La procedura scaricherà e avvierà in background un container chiamato home-assistant al quale verrà passati parametri come il fuso orario del sistema e la coordinata della directory prima creata, la quale verrà adottata come /config all’interno del container quale directory contenente [la configurazione dell’HUB](https://indomus.it/guide/configuration-yaml-capire-il-file-di-configurazione-di-home-assistant/). Il comando, inoltre, comunica al container di avviarsi in automatico quando e se verrà riavviato il servizio Docker.

Dopodiché passa alla parte Primo accesso su Home Assistant.

Mi sono perso qualcosa?

**Reply by snakuzzo** [reply]
Allora...la guida, per quanto accurata possa essere, non devi intenderla come un "manuale di istruzioni" ma deve essere uno spunto per fare le cose. Detto ciò, tu DEVI SAPERE come muoverti se qualcosa non va, altrimenti ogni 10m sarai qui a chiedere info del perché non funziona una cosa e perché non ne funziona un'altra.
Come già ti ho scritto prima... vedi se il container sta girando...

```
docker ps -a
```

Se sta girando, controlla i log di home assistant

```
docker logs -f home-assistant
```

Però considera che qui non è che possiamo fare le lezioni di Docker

**Reply by Al3eRtO** [reply]
Certo sono d'accordo, sto cercando di documentarmi e di imparare per essere autonomo. Se hai dei suggerimenti da darmi in tal senso ti ringrazio in anticipo.
Dopo verifico con i comandi che mi hai indicato

**Reply by snakuzzo** [reply]
ok...fai sapere

**Reply by Al3eRtO** [reply]
CONTAINER ID   IMAGE                                          COMMAND    CREATED        ST              ATUS                            PORTS     NAMES
3b3f6b18535f   ghcr.io/home-assistant/home-assistant:stable   "/init"    14 hours ago   Re              starting (100) 18 seconds ago             home-assistant
858e6cf08c82   hello-world                                    "/hello"   18 hours ago   Ex  

Il log invece

s6-overlay-suexec: fatal: can only run as pid 1

**Reply by snakuzzo** [reply]
Fai così...

```
docker stop home-assistant
docker container rm 3b3f6b18535f
```

e poi ricrea il container "eliminando" l'opzione `--init`

Così...

```
docker run -d --name="home-assistant" -e "TZ=Europe/Rome" -v "/home/deb/homeassistant:/config" -v "/run/dbus:/run/dbus:ro" --net=host --restart always ghcr.io/home-assistant/home-assistant:stable
```

P.S. Se posso darti un consiglio...passa all'utilizzo di docker-compose... diventerà tutto più semplice (trovi le guide sul nostro sito)

**Reply by Al3eRtO** [reply]
Ero indeciso...pensavo che fosse più semplice col run.
Se volessi provare l'altra soluzione cosa dovrei fare avendo già seguito l'altra strada?

**Reply by snakuzzo** [reply]
Sì, ma fai le cose con ordine.
Vedi un attimo se risolvi con docker nudo e crudo.
Poi per passare a compose, ti leggi gli articoli e vedi. Sostanzialmente quello che definisci da riga di comando lo definirai in un file YAML.
Un passo alla volta...e ripeto...leggi prima la documentazione, poi chiedi

**Reply by Al3eRtO** [reply]
ok, grazie

**Reply by snakuzzo** [reply]
...vedrai che ti troverai molto meglio con docker-compose ;)
Fai sapere se risolvi comunque

**Reply by Al3eRtO** [reply]
ho digitato

 docker run -d --name="home-assistant" -e "TZ=Europe/Rome" -v "/home/deb/homeassistant:/config" -v "/run/dbus:/run/dbus:ro" --net=host --restart always ghcr.io/home-assistant/home-assistant:stable"
>
E' fermo così

**Reply by snakuzzo** [reply]
Che vuol dire?

**Comment by Al3eRtO** [comment]
(>) le parentesi sono perchè esce un emoticon altrimenti

**Comment by Al3eRtO** [comment]
Ecco il log

s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun home-assistant (no readiness notification)
s6-rc: info: service legacy-services successfully started
2022-09-06 17:32:14.858 WARNING (MainThread) [homeassistant.config_entries] Config entry 'Bluetooth' for bluetooth integration not ready yet: Failed to start Bluetooth: [org.bluez.Error.InProgress] Operation already in progress; Retrying in background
s6-rc: info: service legacy-services: stopping
Unable to find configuration. Creating default one in /config
[15:37:48] INFO: Home Assistant Core finish process exit code 0
[15:37:48] INFO: Home Assistant Core service shutdown
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped

**Reply by snakuzzo** [reply]
il log non dice nulla...
Prova un attimo a togliere `default_config:` dal file `/home/deb/homeassistant/configuration.yaml` che è stato creato quando hai creato il container.

P.S. ho appena provato sul mio PC e non c'è alcun problema.

**Reply by Al3eRtO** [reply]
Non riesco ad eliminare il file

**Reply by snakuzzo** [reply]
non mi pare di averti detto di eliminare un file

**Reply by Al3eRtO** [reply]
Giusto, scusa. Ho eliminato il default_config ma non me lo fa salvare, non ho i permessi...

**Reply by snakuzzo** [reply]
Perché i file vengono creati come utente `root`.
Usa `sudo nano configurazione.yaml` per editare

**Reply by Al3eRtO** [reply]
Ho fatto ma non cambia niente

**Reply by snakuzzo** [reply]
Credo ci sia qualche problema  a livello di OS, ma così è impossibile (almeno per me) darti una mano. Toccherebbe mettere mano al tuo PC e dare un'occhiata

**Reply by Al3eRtO** [reply]
Non so, ho fatto l'installazione pulita e poi seguito la procedura...
